src/assets/wise5/directives/dynamic-prompt/dynamic-prompt.component.spec.ts

```diff
@@ -75,32 +75,46 @@ function peerGroupingTagDisabled(): void {
 }
 
 function peerGroupingTagEnabled(): void {
+  let retrieveDynamicPromptStudentDataSpy: jasmine.Spy;
   describe('peerGroupingTagEnabled', () => {
     beforeEach(() => {
       fixture = TestBed.createComponent(DynamicPromptComponent);
       component = fixture.componentInstance;
       component.dynamicPrompt = createDynamicPrompt();
       component.dynamicPrompt.peerGroupingTag = 'apple';
       spyOn(TestBed.inject(PeerGroupService), 'retrievePeerGroup').and.returnValue(
-        createPeerGroup()
+        of(createPeerGroup())
+      );
+      retrieveDynamicPromptStudentDataSpy = spyOn(
```

Comment:
I think it's better for readers if we name the spy so they don't have to go find the declaration to figure out what it's spying on.

Reply:
I don't think there's anything wrong with going to the declaration to figure out what it's spying on. When I read the code, I first need to read the declaration code to know what it's spying on, no matter what the variable is called. The result is store in the *Spy variable, and from there it becomes a matter of readability.

I think the longer variable name actually makes the code harder to read because 1. reader has to read more letters each time, and 2. in this case, the statement spans multiple lines, which makes the whole class longer and require scrolling. Renaming this to "retrieveDataSpy" or "studentDataSpy" would still get the meaning across and make it easier to read for me.